src/stcal/testing_helpers.py

```diff
@@ -0,0 +1,36 @@
+import tracemalloc
+
+class MemoryThresholdExceeded(Exception):
+    pass
+
+
+class MemoryThreshold:
+    """
+    Context manager to check peak memory usage against an expected threshold.
+
+    example usage:
+    with MemoryThreshold(expected_usage):
+        # code that should not exceed expected
```

Comment:
```suggestion
    Context manager to check peak memory usage against an expected threshold.

    example usage:
    with MemoryThreshold(expected_usage):
        # code that should not exceed expected
    
    If the code in the with statement uses more than the expected_usage
    memory (in bytes) a ``MemoryThresholdExceeded`` exception
    will be raised.
    
    Note that this class does not prevent allocations beyond the threshold
    and only checks the actual peak allocations to the threshold at the
    end of the with statement.
```
How does this look? I was hoping to add:
- the exception that is raised when the threshold is passed
- a note that this class doesn't cap the memory allocations so code can still allocate more than the threshold (it will just fail at the end if the peak was more than the threshold).

Reply:
seems fine to me